test-cargo-miri/exported-symbol-dep/src/lib.rs

```diff
@@ -2,3 +2,12 @@
 fn exported_symbol() -> i32 {
     123456
 }
+
+pub struct AssocFn;
+
+impl AssocFn {
+    #[no_mangle]
+    pub fn assoc_fn_as_exported_symbol() -> i32 {
```

Comment:
I can't test this in `tests/run-pass`, because `#[no_mangle]` only seems to work on associated functions that are *public* in a *library* crate...
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f1244afcdd26e2a28445f6e82ca46b50

If the associated function is used directly, it can also works with `rustc` when compiling as a binrary, but it still doesn't work in Miri:
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=46fc298b46ff44093c45cab9fddb6650

Reply:
So you are saying even with this patch, https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=46fc298b46ff44093c45cab9fddb6650 fails? Could you open an issue for that?